packages/astro/src/search.ts

```diff
@@ -125,10 +128,11 @@ function loadCollection(url: string, pagesRoot: URL): { currentPage?: number; lo
           }
         }
       }
+      const pagesPath = astroConfig.pages.pathname.replace(astroConfig.projectRoot.pathname, '');
       return {
         location: {
-          fileURL: new URL(`./${pageURL}`, pagesRoot),
-          snowpackURL: `/_astro/pages/${pageURL}.js`,
```

Comment:
TL;DR there was still quite a bit of hardcoded assumptions in Astro about how to resolve URLs. While there are still a few lingering, this removes most of the common ones users may hit and get unexpected “not found” bugs